### PR TITLE
Fixing codec test cases

### LINKED ISSUE #11: Pick a next class to tackle, after Message, do cdefs, start fixing tests, ....

### PR DESCRIPTION
**Pull request by SudhanAruna**
fix #11

### CODE REVIEW

**Comment by jiridanek** [comment]
```
======================================================================
ERROR: testMemoryView (test_codec.DataTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/qpid-proton/qpid-proton/python/tests/proton_tests/test_codec.py", line 506, in testMemoryView
    data.decode(self.data.encode())
  File "/home/runner/work/qpid-proton/qpid-proton/python/proton/_data.py", line 835, in decode
    return self._check(pn_data_decode(self._data, bytes, size[0]))
  File "/home/runner/work/qpid-proton/qpid-proton/python/proton/_data.py", line 690, in _check
    pn_error_text(pn_data_error(self._data))
proton._exceptions.DataException: [-6]: <cdata 'char *' NULL>
```

**Comment by SudhanAruna** [comment]
I'll try to keep code simple like in your commit and updated the doc string for previous file as well.

**Review by jiridanek — CHANGES_REQUESTED** [review]
@SudhanAruna Go over my commit if you agree with the changes, and rebase the PR to resolve the edit conflict in `_data.py` that GitHub highlighted.

File: python/proton/_data.py [hunk 1]
```diff
@@ -686,7 +686,8 @@ def _check(self, err):
             exc = EXCEPTIONS.get(err, DataException)
             raise exc("[%s]: %s" % (
                 err,
-                ffi.string(pn_error_text(pn_data_error(self._data))).decode('utf8')
+                # ffi.string(pn_error_text(pn_data_error(self._data))).decode('utf8')
+                pn_error_text(pn_data_error(self._data))
```
Comment (jiridanek) [inline python/proton/_data.py]: IMO the `.decode()` version looks more likely to be correct here, but I am not sure from just looking at it. I guess you had a reason for the change.

File: python/tests/proton_tests/test_codec.py [hunk 2]
```diff
@@ -410,7 +411,7 @@ def testFloat(self):
     def testDouble(self):
         self._test("double", 0, 1, 2, 3, 0.1, 0.2, 0.3, -1, -2, -3, -0.1, -0.2, -0.3)
 
-    def testBinary(self):
+    def testBinary(self):
```
Comment (jiridanek) [inline python/tests/proton_tests/test_codec.py]: gratuitous whitespace change

File: python/proton/_data.py (hunk 1 above)
Comment (SudhanAruna) [reply python/proton/_data.py]: I commented it purpose of debugging the null pointer issue. It should be left with .decode() for clear error message

**Pull request merged**